app/containers/SongContainer/saga.ts

```diff
@@ -0,0 +1,16 @@
+import { put, call, takeLatest } from 'redux-saga/effects';
+import { requestGetItuneSongs, successGetItuneSongs } from './reducer';
+import { getItunes } from '@app/services/ituneApi';
+
+export function* getItuneSongs(action: any): Generator<any, any> {
+  const res: any = yield call(getItunes, action.payload);
```

Comment:
have your yield call under try catch block so that you can handle any type of request error